src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -94,25 +112,133 @@ static void assumeDocker() throws Exception {
 
     @Rule public LoggerRule logging = new LoggerRule().recordPackage(BourneShellScript.class, Level.FINE);
 
+    private TestPlatform platform;
     private StreamTaskListener listener;
+    private Slave s;
     private FilePath ws;
     private Launcher launcher;
+    private static int counter = 0; // used to prevent docker container name-smashing
+    private boolean testExecuted;
+
+    public BourneShellScriptTest(TestPlatform platform) throws Exception {
+        this.platform = platform;
+        this.listener = StreamTaskListener.fromStdout();
+    }
+
+    @Before public void prepareAgentForPlatform() throws Exception {
+        testExecuted = false;
+        switch (platform) {
+            case NATIVE:
+                testExecuted = true;
+                s = j.createOnlineSlave();
+                break;
+            case SLIM:
+            case ALPINE:
+            case CENTOS:
+            case UBUNTU:
+            case SIMPLE:
+                assumeDocker();
+                testExecuted = true;
+                s = prepareAgentDocker();
+                j.jenkins.addNode(s);
+                j.waitOnline(s);
+                break;
+            default:
+                fail("Unknown enum value: " + platform);
+                break;
+        }
+        ws = s.getWorkspaceRoot().child("ws");
+        launcher = s.createLauncher(listener);
+    }
 
-    @Before public void vars() {
-        listener = StreamTaskListener.fromStdout();
-        ws = j.jenkins.getRootPath().child("ws");
-        launcher = j.jenkins.createLauncher(listener);
+    private Slave prepareAgentDocker() throws Exception {
+        switch(platform) {
+            case SLIM:
+            case ALPINE:
+            case CENTOS:
+            case UBUNTU:
+                return prepareDockerPlatforms();
+            case SIMPLE:
+                return prepareAgentCommandLauncher();
```

Comment:
Not clear to me what `SIMPLE` means, could use Javadoc. Running `jenkins/agent` without `--init`? What about running it _with_ `--init`?

Reply:
Yes, I originally  had two platforms, `SIMPLE` (without --init) and `TINI` (with  `--init`'). I removed `TINI` because the other docker containers have `init` processes, so I felt the coverage was already there.  I'll rename `SIMPLE` to something like`NO_INIT` at the very least.

Reply:
Ah, OK, that would be clearer.